src/fava/core/ingest.py

```diff
@@ -443,16 +447,37 @@ def extract(self, filename: str, importer_name: str) -> list[Directive]:
 
         try:
             path = Path(filename)
+            importer = self.importers[importer_name]
             new_entries = extract_from_file(
-                self.importers[importer_name],
+                importer,
                 path,
                 existing_entries=self.ledger.all_entries,
             )
         except Exception as exc:
             raise ImporterExtractError from exc
 
-        new_entries_list = [(filename, new_entries)]
         for hook_fn in self.hooks:
+            new_entries_list: HookOutput = []
+            try:
+                annotations = get_annotations(hook_fn)
```

Comment:
until today, we actually only had a minimum Python version 3.9, but since beanquery was broken there, I've bumped it to 3.10 so we can use this here as well

Reply:
Removed